node/src/components/consensus/highway_core/finality_detector/rewards.rs

```diff
@@ -136,7 +136,8 @@ fn round_participation<'a, C: Context>(
             .find(|&(_, unit)| unit.round_id() <= r_id),
     };
     maybe_unit.map_or(RoundParticipation::No, |(vh, unit)| {
-        if unit.round_exp > r_id.trailing_zeros() {
+        #[allow(clippy::integer_arithmetic)]
+        if r_id.millis() % unit.round_len.millis() != 0 {
```

Comment:
Maybe add a comment that `round_len` cannot be 0 in the unit, because it is still defined via the exponent in the `WireUnit`s that are sent over the network? So that a malicious validator cannot make it be 0.